File: SpringBoot/src/main/java/com/hospital/qlbv/entity/PrescriptionDetail.java
package com.hospital.qlbv.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "prescription_details")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PrescriptionDetail {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer detailId;
    
    @Column(name = "prescription_id")
    private Integer prescriptionId;
    
    @Column(name = "medicine_id")
    private Integer medicineId;
    
    private String dosage;
    private String quantity;
    
    @Column(name = "usage_instructions")
    private String usageInstructions;
    
    private Integer duration;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "prescription_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"details"})
    private Prescription prescription;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "medicine_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"prescriptionDetails"})
    private Medicine medicine;
}

File: Backend/Models/PrescriptionDetail.cs
using System.ComponentModel.DataAnnotations;

namespace QuanLyBenhVien.API.Models;

public class PrescriptionDetail
{
    [Key]
    public int DetailID { get; set; }
    public int? PrescriptionID { get; set; }
    public int? MedicineID { get; set; }
    public string? Dosage { get; set; }
    public string? Quantity { get; set; }
    public string? UsageInstructions { get; set; }
    public int? Duration { get; set; }

    // Navigation properties
    public virtual Prescription? Prescription { get; set; }
    public virtual Medicine? Medicine { get; set; }
}
